File: jvm/cucumber-blueprint-jvm-example/src/test/java/cucumber/blueprint/jvm/example/integration/cucumber/step/exampleroute/ExampleRouteStepDefinitions.java
package cucumber.blueprint.jvm.example.integration.cucumber.step.exampleroute;

import cucumber.api.java.en.Then;
import cucumber.api.java.en.When;
import cucumber.blueprint.jvm.example.integration.cucumber.step.common.BaseStepDefinition;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.hamcrest.CoreMatchers.notNullValue;
import static org.hamcrest.MatcherAssert.assertThat;

public class ExampleRouteStepDefinitions extends BaseStepDefinition {
    private String result;

    @When("call the example route")
    public void callRest() throws Exception {
        assertThat(camelContext, notNullValue());
        assertThat(exampleOsgiService, notNullValue());
        result = producerTemplate.requestBody("direct:test", "", String.class);
    }

    @Then("the response is (.+)")
    public void verifyExampleRouteResponse(final String response) {
        assertThat(result, equalTo(response));
    }
}

File: scala/cucumber-blueprint-scala-example/src/test/scala/cucumber/blueprint/jvm/example/integration/cucumber/step/exampleroute/ExampleRouteStepDefinitions.scala
package cucumber.blueprint.jvm.example.integration.cucumber.step.exampleroute

import cucumber.runtime.java.blueprint.BlueprintDescriptorLocation
import org.hamcrest.MatcherAssert._
import org.hamcrest.CoreMatchers._
import cucumber.api.java.en.{Then, Given}
import cucumber.blueprint.jvm.example.integration.cucumber.step.common.BaseStepDefinition

@BlueprintDescriptorLocation(value = "/OSGI-INF/blueprint/blueprintContext.xml")
class ExampleRouteStepDefinitions extends BaseStepDefinition {

  @Given("^call the example route$")
  def callRoute() {
    assertThat(producerTemplate, notNullValue())
    result = producerTemplate.requestBody("direct:test", "", classOf[String])
  }

  @Then("^the response is (.+)$")
  def verifyExampleRouteResponse(expected: String) {
    assertThat(result, equalTo(expected))
  }
}